packages/storage-plugin/src/internals.ts

```diff
@@ -70,10 +70,52 @@ export function engineFactory(
   }
 
   if (options.storage === StorageOption.LocalStorage) {
-    return localStorage;
+    try {
+      const verifyKey = `${DEFAULT_STATE_KEY}-VERIFY`;
+      localStorage.setItem(verifyKey, '1');
+      localStorage.removeItem(verifyKey);
+      return localStorage;
+    } catch (e) {
+      console.warn('LocalStorage is not available, using a fallback implementation!', e);
+      return fallbackStorage();
+    }
   } else if (options.storage === StorageOption.SessionStorage) {
-    return sessionStorage;
+    try {
+      const verifyKey = `${DEFAULT_STATE_KEY}-VERIFY`;
+      sessionStorage.setItem(verifyKey, '1');
+      sessionStorage.removeItem(verifyKey);
+      return sessionStorage;
+    } catch (e) {
+      console.warn('SessionStorage is not available, using a fallback implementation!', e);
+      return fallbackStorage();
+    }
```

Comment:
Seems like you have duplicated try-catch statements, you can simplify it:
```ts
const storage =
  options.storage === StorageOption.LocalStorage
    ? localStorage
    : options.storage === StorageOptions.SessionStorage
    ? sessionStorage
    : null;

if (storage === null) {
  return null;
}

try {
  const verifyKey = `${DEFAULT_STATE_KEY}-VERIFY`;
  storage.setItem(verifyKey, '1');
  storage.removeItem(verifyKey);
  return storage;
} catch (error) {
  // Caretaker note: we have still left the `typeof` condition in order to avoid
  // creating a breaking change for projects that still use the View Engine.
  if (typeof ngDevMode === 'undefined' || ngDevMode) {
    console.warn(
      `${
        options.storage === StorageOption.LocalStorage
          ? 'localStorage'
          : 'sessionStorage'
      } is not available, using a fallback implementation!`,
      error
    );
  }
  return fallbackStorage();
}
```

Reply:
thanks @arturovt. 
I don't think this suggestion makes the code easier for me, but I added your suggestion. (probably only a method with try catch with a param of the store engine would have been something I should have added from the start).  Anyhow, small details 😄 

Can I ask, why are we hiding the warning there sometimes? I think we should always print a message.